### ISSUE TITLE
Layer - add method to get parent dataset

### ISSUE THREAD

**Issue by shawnlaffan** [ajolma/Geo-GDAL-FFI#23]
What do you think of adding a method to get at a layer's parent dataset?  

It is not part of the GDAL API so far as I can see, but since Geo::GDAL::FFI stores a hash of parent refs it should be possible to access it and return the relevant item.  

This would be convenient when passing layers between subs to use for selections, as one does not need to also pass the dataset ref.  

Something like this might be sufficient:

```perl
sub GetParentDataset {
    my ($self) = @_;
    return $Geo::GDAL::FFI::parent ($$self);
}
```

**Comment by ajolma** [comment]
I think it is useful.

**Comment by shawnlaffan** [comment]
Thanks.  I'll work it into a PR.

**Comment by shawnlaffan** [comment]
Fixed by PR #24
